src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -90,12 +90,28 @@ unsigned uct_rc_mlx5_iface_progress(void *arg)
 
 static ucs_status_t uct_rc_mlx5_iface_query(uct_iface_h tl_iface, uct_iface_attr_t *iface_attr)
 {
-    uct_rc_iface_t *iface = ucs_derived_of(tl_iface, uct_rc_iface_t);
+    uct_rc_mlx5_iface_t *iface = ucs_derived_of(tl_iface, uct_rc_mlx5_iface_t);
+    uct_rc_iface_t *rc_iface   = &iface->super;
+    size_t max_inline;
+    size_t max_put_inline;
     ucs_status_t status;
 
-    status = uct_rc_iface_query(iface, iface_attr,
-                                UCT_IB_MLX5_PUT_MAX_SHORT(0),
-                                UCT_IB_MLX5_AM_MAX_SHORT(0),
+#if HAVE_IBV_EXP_DM
+    max_inline     = iface->mlx5_common.dm.dm ? ucs_max(iface->mlx5_common.dm.dm->seg_len,
```

Comment:
maybe, for simplicity, set max_inline/max_put_inline first as if there is no DM, and then inside #ifdef check if iface->mlx5_common.dm.dm!=NULL and then override  the values with iface->mlx5_common.dm.dm->seg_len

Reply:
done